tests/test_yaml/stdlib_arrow_after_meta.yaml

```diff
@@ -0,0 +1,873 @@
+    - {{ compiler("cxx") }}
+    - {{ compiler("cuda") }}                 # [cuda_compiler_version != "None"]
+    # needs to run protoc & grpc_cpp_plugin
+    - libgrpc                                # [build_platform != target_platform]
+    - libprotobuf                            # [build_platform != target_platform]
+    # needed for gandiva
+    - clangdev {{ llvm_version }}            # [build_platform != target_platform]
+    - llvmdev {{ llvm_version }}             # [build_platform != target_platform]
+    - gnuconfig                              # [build_platform != target_platform]
+    - cmake
+    - ninja
+    # necessary for vendored jemalloc
+    - autoconf     # [linux]
+    - make         # [linux]
+  host:
+    - {{ stdlib("c") }}
+    # for required dependencies, see
+    # https://github.com/apache/arrow/blob/apache-arrow-11.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake#L46-L75
+    - clangdev {{ llvm_version }}
+    - llvmdev {{ llvm_version }}
+    - aws-crt-cpp
+    - aws-sdk-cpp
+    - brotli
+    - bzip2
+    # not yet: https://github.com/conda-forge/cpp-opentelemetry-sdk-feedstock/issues/38
+    # - cpp-opentelemetry-sdk
+    # - proto-opentelemetry-proto =={{ cpp_opentelemetry_sdk }}
+    - gflags
+    - glog
+    - google-cloud-cpp
+    # arrow uses a customized jemalloc, see #944
+    # - jemalloc
+    - libabseil
+    - libboost-headers
+    - libgrpc
+    - libutf8proc
+    - lz4-c
+    - nlohmann_json
+    - orc
+    - rapidjson
+    - re2
+    - snappy
+    - thrift-cpp
+    - ucx          # [linux]
+    - xsimd
+    - zlib
+    - zstd
+
+outputs:
+  - name: apache-arrow-proc
+    version: {{ build_ext_version }}
+    build:
+      number: {{ proc_build_number }}
+      string: {{ build_ext }}
+    requirements:
+      host:
+        - {{ stdlib("c") }}
+      run_constrained:
+        # avoid installation with old naming of proc package
+        - arrow-cpp-proc <0.0a0
+    test:
+      commands:
+        - exit 0
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: A meta-package to select Arrow build variant
+
+  - name: libarrow-all
+    script: install-libarrow.sh  # [unix]
+    script: install-libarrow.bat  # [win]
+    version: {{ version }}
+    build:
+      string: h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ build_ext }}
+      run_exports:
+        - {{ pin_subpackage("libarrow", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-acero", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-dataset", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-flight", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-flight-sql", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-gandiva", max_pin="x") }}
+        - {{ pin_subpackage("libarrow-substrait", max_pin="x") }}
+        - {{ pin_subpackage("libparquet", max_pin="x") }}
+    requirements:
+      build:
+        - cmake
+        - ninja
+      host:
```

Comment:
Why did this host section not get a stdlib expression? I cannot tell from the logic above.

Reply:
The logic is essentially: If an output does not have its own build section, we assume it inherits from the global build section. However, if an output _does_ have a build section _and_ it contains no compiler (the case here), then we don't need to inject stdlib.

I think that line of thinking is sound (and the result for arrow is arguably correct, even though it's a bit overkill for `apache-arrow-proc`), but of course we can come up with smarter/different logic if we want.